src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+
+  const timeout = setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 && !resolved) {
+      clearTimeout(timeout);
+      resolved = true;
+      resolve('First promise was resolved');
+    }
+  });
```

Comment:
This event listener is used to resolve the promise when the user clicks the left mouse button. But it's not necessary to check if the promise is resolved or not. Because if the promise is resolved, it will not be able to be resolved again. The Promise object will ignore any other attempts to resolve after the first call. So, you can remove the check `!resolved` and the variable `resolved`.